File: api/src/main/java/net/impactdev/gts/api/events/placeholders/PlaceholderReadyEvent.java
package net.impactdev.gts.api.events.placeholders;

import net.impactdev.impactor.api.event.ImpactorEvent;
import org.spongepowered.api.util.annotation.eventgen.GenerateFactoryMethod;

import java.util.UUID;

/**
 * Represents the event when a placeholder's contents becomes available when they previously
 * weren't. This is namely meant for asynchronous placeholder value replacements.
 */
@GenerateFactoryMethod
public interface PlaceholderReadyEvent extends ImpactorEvent {

    UUID getSource();

    String getPlaceholderID();

    Object getValue();

}

File: api/src/main/kotlin/net/impactdev/gts/api/events/placeholders/PlaceholderReadyEvent.kt
package net.impactdev.gts.api.events.placeholders

import net.impactdev.impactor.api.event.ImpactorEvent
import java.util.*

/**
 * Represents the event when a placeholder's contents becomes available when they previously
 * weren't. This is namely meant for asynchronous placeholder value replacements.
 */
interface PlaceholderReadyEvent : ImpactorEvent {
    @get:Param(0)
    val source: UUID

    @get:Param(1)
    val placeholderID: String

    @get:Param(2)
    val value: Any
}